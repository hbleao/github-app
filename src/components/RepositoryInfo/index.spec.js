import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { RepositoryInfo } from '.';

describe('RepositoryInfo', () => {
  it('should be able to render', () => {
    expect(RepositoryInfo).toBeInstanceOf(Function);
  });

  it('should be able to render component with property value', () => {
    render(<RepositoryInfo value="1808" description="Stars" />);

    const valueProperty = screen.getByText('1808');
    expect(valueProperty).toBeTruthy();
  });

  it('should be able to render component with property description', () => {
    render(<RepositoryInfo value="1808" description="Stars" />);

    const descriptionProperty = screen.getByText('Stars');
    expect(descriptionProperty).toBeTruthy();
  });

  it('should be able to render snapshot', () => {
    const component = renderer.create(
      <RepositoryInfo value="1808" description="Stars" />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
