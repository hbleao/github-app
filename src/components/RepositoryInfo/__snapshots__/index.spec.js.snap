// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryInfo should be able to render snapshot 1`] = `
<div
  className="RepositoryInfo"
>
  <p
    className="RepositoryInfo__value"
  >
    1808
  </p>
  <span
    className="RepositoryInfo__description"
  >
    Stars
  </span>
</div>
`;
